src/movies.js

```diff
@@ -1,16 +1,58 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
-
+function getAllDirectors(arr) {
+  const pickDirector = arr.map((x) => x.director);
+  return pickDirector;
+}
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(arr) {
+  const sortedMovie = arr.filter((movie) => {
+    if (
+      movie.director === 'Steven Spielberg' &&
+      movie.genre.indexOf('Drama') > -1
+    ) {
+      return true;
+    }
+  });
+  return sortedMovie.length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(arr) {
+  const sumScore = arr.reduce((prevVal, curVal) => {
+    return prevVal + curVal.score;
+  }, 0);
```

Comment:
The failing test you mentioned. Inside reduce method, before you sum those two values, you need to check if your movie object has property `score`, in other words if this property exists